health-monitors/kubernetes-object-monitor/pkg/controller/reconciler_test.go

```diff
@@ -0,0 +1,740 @@
+			Message:           "GPU job failed",
+			RecommendedAction: "CONTACT_SUPPORT",
+			ErrorCode:         []string{"GPU_JOB_FAILED"},
+		},
+	}
+
+	setup := setupTestWithCRD(t, []config.Policy{crPolicy}, gpuJobCRD())
+	nodeName := "gpu-test-node"
+	jobName := "test-gpu-job"
+	namespace := "default"
+
+	createNode(t, setup, nodeName, v1.ConditionTrue)
+
+	gpuJob := &unstructured.Unstructured{
+		Object: map[string]any{
+			"apiVersion": "batch.nvidia.com/v1alpha1",
+			"kind":       "GPUJob",
+			"metadata": map[string]any{
+				"name":      jobName,
+				"namespace": namespace,
+			},
+			"spec": map[string]any{
+				"nodeName": nodeName,
+			},
+		},
+	}
+
+	require.NoError(t, setup.k8sClient.Create(setup.ctx, gpuJob))
+
+	require.Eventually(t, func() bool {
+		err := setup.k8sClient.Get(setup.ctx, types.NamespacedName{
+			Name:      jobName,
+			Namespace: namespace,
+		}, gpuJob)
+		return err == nil
+	}, time.Second, 50*time.Millisecond)
+
+	gpuJob.Object["status"] = map[string]any{
+		"state": "Failed",
+	}
+	require.NoError(t, setup.k8sClient.Status().Update(setup.ctx, gpuJob))
+
+	result, err := setup.reconciler.Reconcile(setup.ctx, ctrl.Request{
+		NamespacedName: types.NamespacedName{Name: jobName, Namespace: namespace},
+	})
+	assert.NoError(t, err)
+	assert.Equal(t, ctrl.Result{}, result)
+
+	require.Eventually(t, func() bool {
+		if len(setup.publisher.publishedEvents) != 1 {
+			return false
+		}
+		event := setup.publisher.publishedEvents[0]
+		return event.nodeName == nodeName &&
+			!event.isHealthy &&
+			event.policy.Name == "gpu-job-failed"
+	}, time.Second, 50*time.Millisecond)
+
+	setup.publisher.publishedEvents = []mockPublishedEvent{}
+
+	require.NoError(t, setup.k8sClient.Delete(setup.ctx, gpuJob))
+
+	result, err = setup.reconciler.Reconcile(setup.ctx, ctrl.Request{
+		NamespacedName: types.NamespacedName{Name: jobName, Namespace: namespace},
+	})
+	assert.NoError(t, err)
+	assert.Equal(t, ctrl.Result{}, result)
+
+	require.Eventually(t, func() bool {
+		if len(setup.publisher.publishedEvents) != 1 {
+			return false
+		}
+		event := setup.publisher.publishedEvents[0]
+		return event.nodeName == nodeName &&
+			event.isHealthy &&
+			event.policy.Name == "gpu-job-failed"
+	}, time.Second, 50*time.Millisecond)
+}
+
+func TestReconciler_ColdStart(t *testing.T) {
+	tests := []struct {
+		name              string
+		postRestartAction func(*testing.T, *testSetup, string, *v1.Node)
+		expectEvent       bool
+		expectHealthy     bool
+	}{
+		{
+			name: "unhealthy resource",
+			postRestartAction: func(t *testing.T, s *testSetup, nodeName string, _ *v1.Node) {
+				updateNodeStatus(t, s, nodeName, v1.ConditionFalse)
+			},
+			expectEvent: false,
+		},
+		{
+			name: "healthy resource",
+			postRestartAction: func(t *testing.T, s *testSetup, nodeName string, _ *v1.Node) {
+				updateNodeStatus(t, s, nodeName, v1.ConditionTrue)
+			},
+			expectEvent:   true,
+			expectHealthy: true,
+		},
+		{
+			name: "deleted resource",
```

Comment:
can we add a test for custom resource that was deleted between controller reboots?